stackdriver/storage.go

```diff
@@ -72,7 +73,30 @@ func (s *Storage) Close() error {
 // ApplyConfig updates the state as the new config requires.
 func (s *Storage) ApplyConfig(conf *config.Config) error {
 	var projectId string
-	if value, ok := conf.GlobalConfig.ExternalLabels[ProjectIdLabel]; !ok {
+	var location string
+	var clusterName string
+	el := conf.GlobalConfig.ExternalLabels
+	if md.OnGCE() {
+		if id, err := md.ProjectID(); err == nil {
+			projectId = "projects/" + id
+			if v, ok := el[ProjectIdLabel]; !ok || v == "" {
```

Comment:
These lines could just check for an empty value, as the value is never used, same throughout.

```
if el[ProjectIdLabel] == "" {
```

Reply:
Done.